distributed/tests/test_spill.py

```diff
@@ -226,3 +227,50 @@ def __sizeof__(self):
             buf["a"] = a
 
     assert "Failed to pickle" in logs_bad_key.getvalue()
+
+    b = Bad(size=100)  # this is small enough to fit in memory/fast
+
+    buf["b"] = b
+    assert set(buf.fast) == {"b"}
+
+    c = 100 * "c"
+    with captured_logger(logging.getLogger("distributed.spill")) as logs_bad_key_mem:
+        # This will go to fast and try to kick b out,
+        # but keep b in fast since it's not pickable
+        buf["c"] = c
+
+    assert "Failed to pickle" in logs_bad_key_mem.getvalue()
+    assert set(buf.fast) == {"b", "c"}
+
+
+def test_spillbuffer_oserror(tmpdir):
+    buf = SpillBuffer(str(tmpdir), target=200, max_spill=800)
+
+    a, b, c = "a" * 200, "b" * 100, "c" * 200
+
+    # let's have something in fast and something in slow
+    buf["a"] = a
+    buf["b"] = b
+    assert set(buf.fast) == {"b"}
+    assert set(buf.slow) == {"a"}
+
+    # modify permissions of disk to be read only
+    subprocess.call(["chmod", "-R", "-w", str(tmpdir)])
+
+    with captured_logger(logging.getLogger("distributed.spill")) as logs_oserror:
+        buf["c"] = c
+
+    # PROBLEM: this is not triggering the OSError and even though c ends up in fast because
```

Comment:
@crusaderky I'm seeing a weird behaviour trying to capture the OSError. I noticed that if I have something on slow (let's say 'a') and nothing in fast, and try to add a new key 'b' > spill threshold this triggers the OSError correctly. However, if I have something in fast and I tried the same, it does not trigger the OSError and it says that disk reached capacity even if it hasn't. I'm not quite sure what's happening.

Reply:
@ncclementi what, exactly, says that the disk reached capacity? And how?